src/cockpit/transports.py

```diff
@@ -398,6 +399,18 @@ def _close(self) -> None:
             self._sock.close()
             self._sock = None
 
+    def close(self) -> None:
```

Comment:
Now that we do this here, we can probably avoid doing it from the stream channel, no?

Reply:
I was hoping for that, but unfortunately not. TestActivePages still fails. I think the issue there is that SIGTERMing bash on a PTY doesn't actually close the shell; it needs an EOF on stdin for that (which is what mvollmer's commit does). So, this may cover similar cases, but not bash with pty :/